From a85e72a7a4e982aa3bac5bd15b9dc61961ca71e5 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Tue, 6 Nov 2018 15:19:35 +0100
Subject: [PATCH] [SQUSH][DNM] Merge tag 'android-9.0.0_r16' into lineage-16.0

commit 86719e04c6eb8312ed01e49188dcff967be4da28
Merge: 9a32709 6334c8a
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Mon Aug 20 03:05:18 2018 +0000

    Snap for 4963492 from 6334c8adf80e7f02c1e8a72a55c7e44b685e9d5a to pi-qpr1-release

    Change-Id: I92a6e716621d56ca94fbfc8780dcd4e8b41538e6

commit 6334c8adf80e7f02c1e8a72a55c7e44b685e9d5a
Author: Carmen Jackson <carmenjackson@google.com>
Date:   Thu Aug 16 15:38:51 2018 -0700

    Traceur: Make all notification broadcast intents foreground.

    Some apps produce a lot of broadcast queue latency (for example,
    Dynamite in b/79715842), which is causing delays in notification
    handling for Traceur and other apps that don't send their notification
    broadcast intents at foreground priority. This change adds
    Intent.FLAG_RECEIVER_FOREGROUND to Traceur's broadcast intents, giving
    the notifications their expected foreground priority (the user expects
    to see an immediate response when they tap on each of these
    notifications) and protecting the notifications from being delayed due
    to other apps' broadcast behaviors.

    Bug: 112155897
    Test: atest TraceurUiTests completes successfully.
    Test: The Traceur notification flow still works as expected.
    Test: Without this change, tapping on a trace notification immediately
    after a reboot sometimes resulted in delays. With this change, tapping a
    trace notification after reboot resulted in a prompt response.
    Change-Id: I288229eb9f22ca0bb460ffa8e12309c6add08591

commit 9a3270918dbfd0acd961e755bdbac91fc1cd0088
Author: Bill Yi <byi@google.com>
Date:   Wed Jul 4 22:13:07 2018 -0700

    Import translations. DO NOT MERGE

    Auto-generated-cl: translation import

    Bug: 64712476
    Change-Id: If33e5b9334d06c454896ffbd018c6529ce40b079

Change-Id: I259bd8efa1838a97a76e7eecf5fcc92cdb62a42f
---
 res/values-sw/strings.xml                         | 4 ++--
 src/com/google/android/traceur/AtraceService.java | 1 +
 src/com/google/android/traceur/FileSender.java    | 2 +-
 3 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/res/values-sw/strings.xml b/res/values-sw/strings.xml
index 19b9db8..857f868 100644
--- a/res/values-sw/strings.xml
+++ b/res/values-sw/strings.xml
@@ -6,8 +6,8 @@
     <string name="record_trace" msgid="6416875085186661845">"Rekodi nyayo"</string>
     <string name="trace_debuggable_applications" msgid="7957069895298887899">"Tafuta programu zinazoweza kutatuliwa"</string>
     <string name="categories" msgid="2280163673538611008">"Aina"</string>
-    <string name="restore_default_categories" msgid="5090536794637169521">"Rejesha aina chaguo msingi"</string>
-    <string name="default_categories_restored" msgid="6861683793680564181">"Imerejesha aina chaguo msingi"</string>
+    <string name="restore_default_categories" msgid="5090536794637169521">"Rejesha aina chaguomsingi"</string>
+    <string name="default_categories_restored" msgid="6861683793680564181">"Imerejesha aina chaguomsingi"</string>
     <string name="applications" msgid="521776761270770549">"Maombi"</string>
     <string name="no_debuggable_apps" msgid="4386209254520471208">"Haikupata programu zozote ambazo zinaweza kutatuliwa"</string>
     <string name="buffer_size" msgid="3370134227980578569">"Ukubwa wa hifadhi ya akiba"</string>
diff --git a/src/com/google/android/traceur/AtraceService.java b/src/com/google/android/traceur/AtraceService.java
index 2c0f15f..bd44117 100644
--- a/src/com/google/android/traceur/AtraceService.java
+++ b/src/com/google/android/traceur/AtraceService.java
@@ -81,6 +81,7 @@ public class AtraceService extends IntentService {
         Context context = getApplicationContext();
         Intent stopIntent = new Intent(Receiver.STOP_ACTION,
             null, context, Receiver.class);
+        stopIntent.addFlags(Intent.FLAG_RECEIVER_FOREGROUND);
 
         String title = context.getString(R.string.trace_is_being_recorded);
         String msg = context.getString(R.string.tap_to_stop_tracing);
diff --git a/src/com/google/android/traceur/FileSender.java b/src/com/google/android/traceur/FileSender.java
index d38a1be..630726f 100644
--- a/src/com/google/android/traceur/FileSender.java
+++ b/src/com/google/android/traceur/FileSender.java
@@ -43,7 +43,7 @@ public class FileSender {
         final Uri traceUri = getUriForFile(context, file);
 
         Intent sendIntent = buildSendIntent(context, traceUri);
-        sendIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        sendIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_RECEIVER_FOREGROUND);
 
         final Notification.Builder builder =
             new Notification.Builder(context, Receiver.NOTIFICATION_CHANNEL)
-- 
2.17.1

